ProcessOptimizer/space/space.py

```diff
@@ -1,917 +1,940 @@
+
+        Parameters
+        ----------
+        * `low` [float]:
+            Lower bound (inclusive).
+
+        * `high` [float]:
+            Upper bound (inclusive).
+
+        * `prior` ["uniform" or "log-uniform", default="uniform"]:
+            Distribution to use when sampling random points for this dimension.
+            - If `"uniform"`, points are sampled uniformly between the lower
+              and upper bounds.
+            - If `"log-uniform"`, points are sampled uniformly between
+              `log10(lower)` and `log10(upper)`.`
+
+        * `transform` ["identity", "normalize", optional]:
+            The following transformations are supported.
+
+            - "identity", (default) the transformed space is the same as the
+              original space.
+            - "normalize", the transformed space is scaled to be between
+              0 and 1.
+
+        * `name` [str or None]:
+            Name associated with the dimension, e.g., "learning rate".
+        """
+        if high <= low:
+            raise ValueError(
+                "the lower bound {} has to be less than the"
+                " upper bound {}".format(low, high)
+            )
+        self.low = low
+        self.high = high
+        self.prior = prior
+        self.name = name
+
+        if transform is None:
+            transform = "identity"
+
+        self.transform_ = transform
+
+        if self.transform_ not in ["normalize", "identity"]:
+            raise ValueError(
+                "transform should be 'normalize' or 'identity'"
+                " got {}".format(self.transform_)
+            )
+
+        if self.transform_ == "normalize":
+            # set upper bound to next float after 1. to make the numbers
+            # inclusive of upper edge
+            if self.prior == "uniform":
+                self.transformer = Pipeline([Identity(), Normalize(low, high)])
+            else:
+                self.transformer = Pipeline(
+                    [Log10(), Normalize(np.log10(low), np.log10(high))]
+                )
+        else:
+            if self.prior == "uniform":
+                self.transformer = Identity()
+            else:
+                self.transformer = Log10()
+
+    def __eq__(self, other):
+        return (
+            type(self) is type(other)
+            and np.allclose([self.low], [other.low])
+            and np.allclose([self.high], [other.high])
+            and self.prior == other.prior
+            and self.transform_ == other.transform_
+        )
+
+    def __repr__(self):
+        return "Real(low={}, high={}, prior='{}', transform='{}')".format(
+            self.low, self.high, self.prior, self.transform_
+        )
+
+    def inverse_transform(self, Xt):
+        """Inverse transform samples from the warped space back into the
+        orignal space.
+        """
+        return np.clip(
+            super(Real, self).inverse_transform(Xt).astype(float), self.low, self.high
```

Comment:
```suggestion
            super(Real, self).inverse_transform(Xt).astype(float), 
            self.low, 
            self.high
```
Makes it easier to see that np.clip() is getting three inputs.